<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    <meta name="HandheldFriendly" content="True" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <title>Rhizosphere metatranscriptome analysis  0/n</title>
    <meta name="description" content="" />
    <link href="//fonts.googleapis.com/css?family=Noto+Sans:300,400,700" rel="stylesheet" type="text/css">
    <link href="//fonts.googleapis.com/css?family=Noto+Serif:400,700,400italic" rel="stylesheet" type="text/css">
    <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
    <link href="//jonathandmoore.github.io/themes/Saga/assets/css/style.css?v=1522321766860" rel="stylesheet" type="text/css">
    <link href="//jonathandmoore.github.io/themes/Saga/assets/css/animate.min.css?v=1522321766860" rel="stylesheet" type="text/css">
    <link href="https://jonathandmoore.github.io/favicon.ico" rel="shortcut icon">
    <link rel="canonical" href="https://jonathandmoore.github.io/2016/09/08/Rhizosphere-metatranscriptome-analysis-0n.html" />
    <meta name="referrer" content="origin" />
    
    <meta property="og:site_name" content="Jonathan Moore" />
    <meta property="og:type" content="article" />
    <meta property="og:title" content="Rhizosphere metatranscriptome analysis  0/n" />
    <meta property="og:description" content="We&amp;#8217;ve done some 16S, 18S and ITS analysis on the field samples, and now the first metatranscriptome samples are in library prep for Illumina sequencing, so it&amp;#8217;s time for some fettling. I took a look at the EBI metagenomics portal, and took some inspiration from their" />
    <meta property="og:url" content="https://jonathandmoore.github.io/2016/09/08/Rhizosphere-metatranscriptome-analysis-0n.html" />
    <meta property="article:published_time" content="2016-09-08T00:00:00.000Z" />
    <meta property="article:tag" content="Rhizosphere" />
    <meta property="article:tag" content=" Metatranscriptome" />
    
    <meta name="twitter:card" content="summary" />
    <meta name="twitter:title" content="Rhizosphere metatranscriptome analysis  0/n" />
    <meta name="twitter:description" content="We&amp;#8217;ve done some 16S, 18S and ITS analysis on the field samples, and now the first metatranscriptome samples are in library prep for Illumina sequencing, so it&amp;#8217;s time for some fettling. I took a look at the EBI metagenomics portal, and took some inspiration from their" />
    <meta name="twitter:url" content="https://jonathandmoore.github.io/2016/09/08/Rhizosphere-metatranscriptome-analysis-0n.html" />
    
    <script type="application/ld+json">
{
    "@context": "http://schema.org",
    "@type": "Article",
    "publisher": "Jonathan Moore",
    "author": {
        "@type": "Person",
        "name": "Jonathan Moore",
        "image": "https://avatars1.githubusercontent.com/u/7888298?v=4",
        "url": "https://jonathandmoore.github.io/author/jonathandmoore/",
        "sameAs": "https://jonathandmoore.github.io"
    },
    "headline": "Rhizosphere metatranscriptome analysis  0/n",
    "url": "https://jonathandmoore.github.io/2016/09/08/Rhizosphere-metatranscriptome-analysis-0n.html",
    "datePublished": "2016-09-08T00:00:00.000Z",
    "keywords": "Rhizosphere,  Metatranscriptome",
    "description": "We&amp;#8217;ve done some 16S, 18S and ITS analysis on the field samples, and now the first metatranscriptome samples are in library prep for Illumina sequencing, so it&amp;#8217;s time for some fettling. I took a look at the EBI metagenomics portal, and took some inspiration from their"
}
    </script>

    <meta name="generator" content="HubPress" />
    <link rel="alternate" type="application/rss+xml" title="Jonathan Moore" href="https://jonathandmoore.github.io/rss/" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.10.0/styles/atom-one-dark.min.css">
    
        <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>
</head>
<body class="post-template tag-Rhizosphere tag-Metatranscriptome">
    <header id="header" class="animated fadeIn">
    <div class="header-background">
    <section class="blog-content">
        <a id="site-url" class="blog-title" href="https://jonathandmoore.github.io">Jonathan Moore</a>
        <span class="blog-description">Bio data fettling</span>
        <nav class="links fadeIn animated">
            <ul>
                <li class="rss"><a title="RSS Feed" href="/rss/"><i class="fa fa-fw fa-rss-square"></i></a></li>
        
            </ul>
        </nav>
    </section>
    <section class="header-content">
        <h1 class="post-title animated fadeInUp">Rhizosphere metatranscriptome analysis  0/n</h1>
        <span class="post-data"><span class="date animated fadeInUp"><i class="fa fa-clock-o"></i> <time class="timesince date" data-timesince="1473289200" datetime="2016-09-08T00:00" title="08 September 2016">2 years ago</time></span>
            <span class="tags animated fadeInUp"><i class="fa fa-tags"></i> <a href="https://jonathandmoore.github.io/tag/Rhizosphere/">Rhizosphere</a>, <a href="https://jonathandmoore.github.io/tag/Metatranscriptome/"> Metatranscriptome</a></span>
            <span class="author animated fadeInUp"><i class="fa fa-user"></i> <a href="https://jonathandmoore.github.io/author/jonathandmoore/">Jonathan Moore</a></span></span>
    </section>
    </div>
</header>
<main id="main" class="content">
    <article itemtype="http://schema.org/BlogPosting" class="animated fadeIn content post post tag-Rhizosphere tag-Metatranscriptome">
        <section class="post-content">
            <div class="paragraph">
<p>We&#8217;ve done some 16S, 18S and ITS analysis on the field samples, and now the first metatranscriptome samples are in library prep for Illumina sequencing, so it&#8217;s time for some fettling.</p>
</div>
<div class="paragraph">
<p>I took a look at the EBI metagenomics portal, and took some inspiration from their pipeline: <a href="https://www.ebi.ac.uk/metagenomics/about" class="bare">https://www.ebi.ac.uk/metagenomics/about</a></p>
</div>
<div class="paragraph">
<p>My initial plan for our metatranscriptome analysis goes like this:</p>
</div>
<div class="paragraph">
<p>First the usual <a href="http://earlham.ac.uk" class="bare">http://earlham.ac.uk</a> primary analysis on the data, comprising:</p>
</div>
<div class="literalblock">
<div class="content">
<pre>Demultiplexing with bcl2fastq
FASTQC
Kontaminant screening of a random sample of reads, for likely contaminants
MEGAN screening of a random sample of reads, for a taxonomic overview of the sample</pre>
</div>
</div>
<div class="paragraph">
<p>Providing the quality looks good, we&#8217;ll move on to develop the metatranscriptome analysis.  First a clean-up.</p>
</div>
<div class="literalblock">
<div class="content">
<pre>Trim using Trimmomatic to remove low quality segments and adaptors
Assemble read pairs using PANDAseq (probably)</pre>
</div>
</div>
<div class="paragraph">
<p>Then comes the first fork in the road. We would like to treat ncRNA and mRNA sequence reads differently in the subsequent steps in the analysis.</p>
</div>
<div class="literalblock">
<div class="content">
<pre>Use SortMeRNA to classify sequence reads containing rRNA
   Something to pull out tRNAs</pre>
</div>
</div>
<div class="paragraph">
<p>There are some complexities with this, as the SortMeRNA rRNA database may not be very sensitive for soil organisms, particularly eukaryotes. There is a lot of underexplored genomic dark matter in the soil, so I&#8217;m told, so we may do some experimenting with various alternative rRNA databases.</p>
</div>
<div class="paragraph">
<p>Once we have the ncRNA/mRNA classification of each molecule nailed down, on to the detailed classifications.</p>
</div>
<div class="paragraph">
<p>Using rRNA sequences to do taxonomic classification is relatively well explored territory.  We will most likely stick to a Qiime analysis here, but again, we are likely to want to use a specialised database for classification rather than SILVA or GreenGenes.</p>
</div>
<div class="paragraph">
<p>Classifying mRNA sequences is likely a bit more tricky.</p>
</div>
<div class="paragraph">
<p>I recently read that if you ask three different classification methods to classify a sequence read, you are only likely to get two of them to agree on a genus classification about 60% of the time, and only then on reads of 500nt.  Species-level classification is even worse.  Functional classification is even worse still <a href="http://www.nature.com/articles/srep19233" class="bare">http://www.nature.com/articles/srep19233</a>.</p>
</div>
<div class="paragraph">
<p>WEVote looks promising in this respect, and I think it&#8217;s a rational approach to taxonomic classification, but I haven&#8217;t had a chance to trying implementing it yet: <a href="http://biorxiv.org/content/biorxiv/early/2016/05/22/054205.full.pdf" class="bare">http://biorxiv.org/content/biorxiv/early/2016/05/22/054205.full.pdf</a></p>
</div>
<div class="paragraph">
<p>With that in mind, we will be using a couple of different methods for both taxonomic and functional classification, and correlating the results.  We will also be paying careful attention to the databases we use for the classifications, as this has as much effect, if not more, than the chosen algorithm.</p>
</div>
<div class="paragraph">
<p>More later.</p>
</div>
        </section>

    
        <section class="post-comments">
          <div id="disqus_thread"></div>
          <script type="text/javascript">
          var disqus_shortname = 'bio-data-fettling'; // required: replace example with your forum shortname
          /* * * DON'T EDIT BELOW THIS LINE * * */
          (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
          })();
          </script>
          <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
          <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
        </section>
    
    </article>

</main>
    <footer class="animated fadeIn" id="footer">
        <section class="colophon">
          <section class="copyright">Copyright &copy; <span itemprop="copyrightHolder">Jonathan Moore</span>. <span rel="license">All Rights Reserved</span>.</section>
          <section class="poweredby">Published with <a class="icon-ghost" href="http://hubpress.io">HubPress</a></section>
        </section>
        <section class="bottom">
          <section class="attribution">
            <a href="http://github.com/Reedyn/Saga">Built with <i class="fa fa-heart"></i> and Free and Open-Source Software</a>.
          </section>
        </section>
    </footer>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js?v="></script> <script src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.9.0/moment-with-locales.min.js?v="></script> <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.10.0/highlight.min.js?v="></script> 
      <script type="text/javascript">
        jQuery( document ).ready(function() {
          // change date with ago
          jQuery('ago.ago').each(function(){
            var element = jQuery(this).parent();
            element.html( moment(element.text()).fromNow());
          });
        });

        hljs.initHighlightingOnLoad();
      </script>
    <script src="//jonathandmoore.github.io/themes/Saga/assets/js/scripts.js?v=1522321766860"></script>
    
</body>
</html>
