<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    <meta name="HandheldFriendly" content="True" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <title>Ensemble analysis of Differential Expression from RNA-Seq data sets</title>
    <meta name="description" content="" />
    <link href="//fonts.googleapis.com/css?family=Noto+Sans:300,400,700" rel="stylesheet" type="text/css">
    <link href="//fonts.googleapis.com/css?family=Noto+Serif:400,700,400italic" rel="stylesheet" type="text/css">
    <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
    <link href="//jonathandmoore.github.io/themes/Saga/assets/css/style.css?v=1522321766848" rel="stylesheet" type="text/css">
    <link href="//jonathandmoore.github.io/themes/Saga/assets/css/animate.min.css?v=1522321766848" rel="stylesheet" type="text/css">
    <link href="https://jonathandmoore.github.io/favicon.ico" rel="shortcut icon">
    <link rel="canonical" href="https://jonathandmoore.github.io/2016/09/12/Ensemble-analysis-of-Differential-Expression-from-RNA-Seq-data-sets.html" />
    <meta name="referrer" content="origin" />
    
    <meta property="og:site_name" content="Jonathan Moore" />
    <meta property="og:type" content="article" />
    <meta property="og:title" content="Ensemble analysis of Differential Expression from RNA-Seq data sets" />
    <meta property="og:description" content="Recently I&amp;#8217;ve seen discussions among bioinformatians of which is &#x27;the best&#x27; method for DE analysis of RNA-Seq data.  It&amp;#8217;s a recurring theme, and mostly the answer is &#x27;it depends on context and aims&#x27; although there is also evidence that some methods are just better than others." />
    <meta property="og:url" content="https://jonathandmoore.github.io/2016/09/12/Ensemble-analysis-of-Differential-Expression-from-RNA-Seq-data-sets.html" />
    <meta property="article:published_time" content="2016-09-12T00:00:00.000Z" />
    <meta property="article:tag" content="RNA-Seq" />
    <meta property="article:tag" content=" Ensemble" />
    <meta property="article:tag" content=" Differential Expression" />
    
    <meta name="twitter:card" content="summary" />
    <meta name="twitter:title" content="Ensemble analysis of Differential Expression from RNA-Seq data sets" />
    <meta name="twitter:description" content="Recently I&amp;#8217;ve seen discussions among bioinformatians of which is &#x27;the best&#x27; method for DE analysis of RNA-Seq data.  It&amp;#8217;s a recurring theme, and mostly the answer is &#x27;it depends on context and aims&#x27; although there is also evidence that some methods are just better than others." />
    <meta name="twitter:url" content="https://jonathandmoore.github.io/2016/09/12/Ensemble-analysis-of-Differential-Expression-from-RNA-Seq-data-sets.html" />
    
    <script type="application/ld+json">
{
    "@context": "http://schema.org",
    "@type": "Article",
    "publisher": "Jonathan Moore",
    "author": {
        "@type": "Person",
        "name": "Jonathan Moore",
        "image": "https://avatars1.githubusercontent.com/u/7888298?v=4",
        "url": "https://jonathandmoore.github.io/author/jonathandmoore/",
        "sameAs": "https://jonathandmoore.github.io"
    },
    "headline": "Ensemble analysis of Differential Expression from RNA-Seq data sets",
    "url": "https://jonathandmoore.github.io/2016/09/12/Ensemble-analysis-of-Differential-Expression-from-RNA-Seq-data-sets.html",
    "datePublished": "2016-09-12T00:00:00.000Z",
    "keywords": "RNA-Seq,  Ensemble,  Differential Expression",
    "description": "Recently I&amp;#8217;ve seen discussions among bioinformatians of which is &#x27;the best&#x27; method for DE analysis of RNA-Seq data.  It&amp;#8217;s a recurring theme, and mostly the answer is &#x27;it depends on context and aims&#x27; although there is also evidence that some methods are just better than others."
}
    </script>

    <meta name="generator" content="HubPress" />
    <link rel="alternate" type="application/rss+xml" title="Jonathan Moore" href="https://jonathandmoore.github.io/rss/" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.10.0/styles/atom-one-dark.min.css">
    
        <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>
</head>
<body class="post-template tag-RNA-Seq tag-Ensemble tag-Differential-Expression">
    <header id="header" class="animated fadeIn">
    <div class="header-background">
    <section class="blog-content">
        <a id="site-url" class="blog-title" href="https://jonathandmoore.github.io">Jonathan Moore</a>
        <span class="blog-description">Bio data fettling</span>
        <nav class="links fadeIn animated">
            <ul>
                <li class="rss"><a title="RSS Feed" href="/rss/"><i class="fa fa-fw fa-rss-square"></i></a></li>
        
            </ul>
        </nav>
    </section>
    <section class="header-content">
        <h1 class="post-title animated fadeInUp">Ensemble analysis of Differential Expression from RNA-Seq data sets</h1>
        <span class="post-data"><span class="date animated fadeInUp"><i class="fa fa-clock-o"></i> <time class="timesince date" data-timesince="1473634800" datetime="2016-09-12T00:00" title="12 September 2016">2 years ago</time></span>
            <span class="tags animated fadeInUp"><i class="fa fa-tags"></i> <a href="https://jonathandmoore.github.io/tag/RNA-Seq/">RNA-Seq</a>, <a href="https://jonathandmoore.github.io/tag/Ensemble/"> Ensemble</a>, <a href="https://jonathandmoore.github.io/tag/Differential-Expression/"> Differential Expression</a></span>
            <span class="author animated fadeInUp"><i class="fa fa-user"></i> <a href="https://jonathandmoore.github.io/author/jonathandmoore/">Jonathan Moore</a></span></span>
    </section>
    </div>
</header>
<main id="main" class="content">
    <article itemtype="http://schema.org/BlogPosting" class="animated fadeIn content post post tag-RNA-Seq tag-Ensemble tag-Differential-Expression">
        <section class="post-content">
            <div class="paragraph">
<p>Recently I&#8217;ve seen discussions among bioinformatians of which is 'the best' method for DE analysis of RNA-Seq data.  It&#8217;s a recurring theme, and mostly the answer is 'it depends on context and aims' although there is also evidence that some methods are just better than others.</p>
</div>
<div class="paragraph">
<p>For the past couple of years, my answer has been that the best method is 'all of them'.  I&#8217;ve come round to thinking that an ensemble of methods probably gives the best result most of the time.</p>
</div>
<div class="paragraph">
<p>This isn&#8217;t really a new problem or a new question.  Before RNA-Seq was <em>de rigeur</em> for expression experiments, we noticed that in microarray experiments, the list of DE genes varied depending on the method used to estimate DE.  In our projects at the time, the 'best' method for DE analysis was a source of regular debate in project meetings among bioinformaticians, or statisticians, or modellers.</p>
</div>
<div class="paragraph">
<p>We came to the conclusion that the most robust approach was to use multiple methods, and to correlate them.  For example, in this timeseries analysis <a href="http://www.plantcell.org/content/24/9/3530.full" class="bare">http://www.plantcell.org/content/24/9/3530.full</a>, we used two different methods for estimating whether genes were DE.  We then did a manual (or is that visual?) review of the expression profiles of the anomalous cases where only one of the methods score the gene as significantly DE.  This review led us to a conclusion about which method was more reliable in the edge cases for this particular data set.</p>
</div>
<div class="paragraph">
<p>Fast forward a bit and the same issue is live in RNA-Seq expression analysis.  You&#8217;ve aligned your reads or analysed your kmer content, and have a score or count for each gene or transcript, and you need to decide which tool to use to estimate DE.</p>
</div>
<div class="paragraph">
<p>This Venn diagram from Chen et al, 2015 was recently used to give a nice example of the problem:</p>
</div>
<div class="imageblock">
<div class="content">
<img src="http://static-content.springer.com/image/art%3A10.1186%2F1471-2164-16-S7-S14/MediaObjects/12864_2015_Article_7208_Fig6_HTML.jpg" alt="12864 2015 Article 7208 Fig6 HTML.jpg">
</div>
</div>
<div class="paragraph">
<p>Each method comes up with a different list of genes that are DE. There is some overlap between methods, but it&#8217;s not immediately clear which method is 'right' or 'wrong' about which particular genes.</p>
</div>
<div class="paragraph">
<p>My answer is to believe all the methods, to some extent, and to take an ensemble of methods as my estimate of DE.</p>
</div>
<div class="paragraph">
<p><strong>The problem is how to take an ensemble of DE analysis methods.</strong></p>
</div>
<div class="paragraph">
<p>My insight, and my gut feeling, was that when we take the Venn diagram/p-value cut-off approach, then we throw away most of the information about DE.  The tyranny of p&lt;0.05 demands that genes are DE or not.  In reality, there is a whole spectrum from 'almost definitely not DE at all' to 'almost definitely DE'.  The analysis tools even make it easy for us by giving an output of q-values, or FDR values for each gene or transcript, that are essentially a metric of 'DE-ness'.</p>
</div>
<div class="paragraph">
<p>I decided to combine all these separate metrics into a single, ensemble, metric of DE-ness. A natural fettling approach suggested itself:</p>
</div>
<div class="literalblock">
<div class="content">
<pre>Use each tool to generate a ranked order list of all genes,
       sorted by q-value or FDR.
For each gene, calculate its sum of ranks,
       by adding up the ranks each individual method had given the gene.
Sort the gene list by this rank-sum metric.</pre>
</div>
</div>
<div class="paragraph">
<p>This gave me a list of genes, sorted from most likely to be DE, as estimated by all the methods, to least likely.  Bingo, I have an ensemble DE metric.</p>
</div>
<div class="paragraph">
<p>The only thing that remains is to answer the biologist&#8217;s key question "Which genes are DE?".  How to choose where to cut-off the list?</p>
</div>
<div class="paragraph">
<p>My view on cut-offs is to cut the list to suit my needs.</p>
</div>
<div class="paragraph">
<p>If you want a strict cut-off, because you are going to spend money on follow-up qPCR experiments, then take a cutoff where all genes above the cut-off are assessed as significant by all methods.  If you want a lax cut-off, because you want sensitivity, and you are going to follow up with a cheap bioinformatic method like functional enrichment testing, then choose the cutoff accordingly - cut-off at the point where all genes above the cutoff were found significant by at least one method, or at least two.</p>
</div>
<div class="paragraph">
<p>I think there may also be principled ways to come up with a cut-off.</p>
</div>
<div class="paragraph">
<p>One would expect that there is a region at the significant end of the list, where the rank correlation between methods is high, and a region at the non-significant end of the list where rank correlation between methods is low to non-existent.  One could try to detect the zone where the rank correlation starts to decay to an unacceptable degree.</p>
</div>
<div class="paragraph">
<p>A different method of choosing a principled cutoff might be through some kind of leave-one-out cross-validation approach.</p>
</div>
<div class="paragraph">
<p>This ensemble approach is flexible and extensible to as many different methods of DE assessment as you can choose.  My initial choices might be DESeq2, EdgeR, BaySeq, cuffdiff, and limma-voom, because they all have such different models of what it means to be DE, so are likely to form an interesting ensemble.  You might prefer your own or different methods.</p>
</div>
<div class="paragraph">
<p>I think there is a right answer to "Which methods should I use in my ensemble?" and I think it is "All of them".</p>
</div>
<div class="paragraph">
<p>The point is, all methods tend to have a sweet spot where they are particularly sensitive and/or specific, so the more methods you use in your ensemble, the more potential you have for a generally good ensemble.</p>
</div>
<div class="paragraph">
<p>In our first public outing for this method (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3915549/" class="bare">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3915549/</a>) we used three different methods of DE assessment, and we cutoff at the point where at least two methods classified a gene as significant.  Our approach was queried by one of our anonymous reviewers.  They asked whether this was an accepted method, and whether we could point to evidence that it would work.  We responded with logic, rather than direct evidence.  We showed that we were using a number of well-cited accepted methods, and correlating their results for robustness.  We also showed that the burgeoning fields of Data Science, and competitive mathematical modelling, were rich with examples of ensemble methods outperforming a single method.  Our argument was accepted.</p>
</div>
<div class="paragraph">
<p>The obvious extension of the approach is to start weighting the methods in the ensemble, either based on prior information about how good they are, or where their sweet spots are, or using an iterative approach based on the data in hand.  I might get back to that later.</p>
</div>
<div class="paragraph">
<p>Have you used an ensemble method for estimating differential gene expression?  Do you think this is a worthwhile approach?</p>
</div>
        </section>

    
        <section class="post-comments">
          <div id="disqus_thread"></div>
          <script type="text/javascript">
          var disqus_shortname = 'bio-data-fettling'; // required: replace example with your forum shortname
          /* * * DON'T EDIT BELOW THIS LINE * * */
          (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
          })();
          </script>
          <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
          <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
        </section>
    
    </article>

</main>
    <footer class="animated fadeIn" id="footer">
        <section class="colophon">
          <section class="copyright">Copyright &copy; <span itemprop="copyrightHolder">Jonathan Moore</span>. <span rel="license">All Rights Reserved</span>.</section>
          <section class="poweredby">Published with <a class="icon-ghost" href="http://hubpress.io">HubPress</a></section>
        </section>
        <section class="bottom">
          <section class="attribution">
            <a href="http://github.com/Reedyn/Saga">Built with <i class="fa fa-heart"></i> and Free and Open-Source Software</a>.
          </section>
        </section>
    </footer>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js?v="></script> <script src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.9.0/moment-with-locales.min.js?v="></script> <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.10.0/highlight.min.js?v="></script> 
      <script type="text/javascript">
        jQuery( document ).ready(function() {
          // change date with ago
          jQuery('ago.ago').each(function(){
            var element = jQuery(this).parent();
            element.html( moment(element.text()).fromNow());
          });
        });

        hljs.initHighlightingOnLoad();
      </script>
    <script src="//jonathandmoore.github.io/themes/Saga/assets/js/scripts.js?v=1522321766848"></script>
    
</body>
</html>
