= Rhizosphere metatranscriptome analysis

I'm a co-investigator on the BBSRC/NERC-funded Roots of Decline project looking at the interactions between crop rotations, rhizosphere microbiome and oilseed rape crop yield.  The project has been tweeting progress https://twitter.com/BBSRC_SARISA and it's been interesting to see the combination of experimental plots, field sampling from farms, and lab work.  

We've done some 16S, 18S and ITS analysis on the field samples, and now the first metatranscriptome samples are in library prep for Illumina sequencing, so it's time for some fettling.

I took a look at the EBI metagenomics portal, and took some inspiration from their pipeline: https://www.ebi.ac.uk/metagenomics/about


My initial plan for our metatranscriptome analysis goes something like this:

First the usual http://earlham.ac.uk primary analysis on the data, comprising:

	Demultiplexing with bcl2fastq
	FASTQC
	Kontaminant screening of a random sample of reads, for likely suspects
	MEGAN screening of a random sample of reads, for a taxonomic overview of the sample
	
Providing the quality looks good, we'll move on to develop the metatranscriptome analysis.  First a clean-up.

	Trim using Trimmomatic to remove low quality segments and adaptors
	Assemble read paris using PANDAseq (probably)

Then comes the first fork in the road. We would like to treat rRNA and mRNA sequence reads differently in the subsequent steps in the analysis.

	Use SortMeRNA to classify sequence reads as rRNA and mRNA

There are some complexities with this, as the SortMeRNA rRNA database may not be very sensitive for soil organisms, particularly eukaryotes. There is a lot of underexplored genomic dark matter in the soil, so I'm told, so we may do some experimenting with various alternative rRNA databases.

Once we have the rRNA/mRNA classification of each molecule nailed down, on to the detailed classifications.

Using rRNA sequences to do taxonomic classification is relatively well explored territory.  We will most likely stick to a Qiime analysis here, but again, we are likely to want to use a specialised database for classification rather than SILVA or GreenGenes.

Classifying mRNA sequences is a bit more tricky.

I recently read that if you ask three different classification methods to classify a sequence read, you are only likely to get two of them to agree on a genus classification about 60% of the time, and only then on reads of 500nt.  Species-level classification is even worse.  Functional classification is even worse still http://www.nature.com/articles/srep19233.

WEVote looks promising in this respect, and I think it's a rational approach to taxonomic classification, but I haven't had a chance to trying implementing it yet: http://biorxiv.org/content/biorxiv/early/2016/05/22/054205.full.pdf

With that in mind, we will be using a couple of different methods for both taxonomic and functional classification, and correlating the results.  We will also be paying careful attention to the databases we use for the classifications, as this has as much effect, if not more, than the chosen algorithm.

More later.

